Upgrade navbar

BeaRoodriguees opened the pull request:
# Proposed Changes
- Mudança da componente da Navbar

# Screenshots
Navbar padrão:
![image](https://github.com/user-attachments/assets/a4ec0f2e-6aae-401e-ad8b-80f26db1693f)

Navbar apenas com a logo:
![image](https://github.com/user-attachments/assets/767ea3ef-cbae-434b-9690-2e2a3ead65ff)

Navbar da homepage:
![image](https://github.com/user-attachments/assets/43f2df0b-d20c-4f32-89db-f34da914cc88)

# Ajustes
- [x] Logo está com um Anchor para a '/problemset', mas quando está deslogado, gostaria de que fosse para '/'

BeaRoodriguees committed:
feat: create navbar

BeaRoodriguees committed:
style: fix height content of 404 page

BeaRoodriguees committed:
style: format files

Arthurpmrs commented on ui/src/app/(auth)/not-found/page.tsx:
```diff
@@ -1,10 +1,15 @@
-    <div className={classes.container}>
-      <NotFound></NotFound>
-    </div>
+    <>
+      <Navbar status={NavbarStatus.AUTH}></Navbar>
+      <div className={classes.container}>
+        <NotFound></NotFound>
+      </div>
+    </>
```
Acho que aqui poderia ser aquele element `<main></main>` igual você fez para a página de login, achei adequado.

Arthurpmrs commented on ui/src/components/navbar/navbar.tsx:
Alguns comentários e sugestões:

1. Quando eu estou logado e acesso a home page, os botões de Entrar e Registrar estão visíveis. Acho que seria melhor ser igual à navbar do problemset.
2. Acho que poderia aumentar o espaçamento entre "Problemas Turmas", search bar e a foto. Além disso, aumentar o constraste entre o link da página ativa e o outro.
3. Acho que seria bom extrair as opções em dois componentes separados (funções no mesmo arquivo), um com os links, searchbar e foto, e outro com os botões de autenticação, poderiam se chamar algo do tipo "NavbarAuthOptions" e "NavbarLoggedOptions".

Por último, só uma sugestão. Essa é mais questão de preferência, não precisa se preocupar tanto. Ao invés de usar o switch, uma opção é fazer algo desse tipo:
```jsx
{status === NavbarStatus.HOME && <NavbarAuthOptions />}
{status === NavbarStatus.LOGGED && <NavbarLoggedOptions />}
```
Somente uma sugestão.

BeaRoodriguees replied:
1- Como eu consigo pegar a informação que o usuário está logado?

3 - Acho que não entendi como fazer isso, mas vou procurar

4 - Não sabia da existência disso, vou pesquisar sobre e atualizar!

Arthurpmrs replied:
1 - Para verificar se o usuário está logado, você usa essa função aqui:

```js
import { useSession } from 'next-auth/react';
const session = useSession();
const user = session.data?.user
```
Se user for `undefined` o usuário não está logado. Caso contrário vai ser um objeto com as credenciais.

3 - Seria algo desse tipo:

```jsx
function ASubComponent() {
  returnn <button>Opa</button>
}


export default function Navbar() {
   return (
     <div>
       <ASubComponent />
     </div>
  )
}
```

4 - Blzz

BeaRoodriguees committed:
style: fix gap menu

BeaRoodriguees committed:
fix: changes requested in the review

Arthurpmrs committed:
Delete ui/.python-version

Acho que esse arquivo veio de uma tentativa de rodar o backend mas estando na pasta ui, ao invés de api.

Arthurpmrs reviewed (approved):
Aproovad

daviromao commented on ui/src/app/(auth)/register/page.tsx:
```diff
@@ -1,5 +1,12 @@
 export default function RegisterPage() {
-  return <RegistrationForm></RegistrationForm>;
+  return (
+    <main>
+      <Navbar status={NavbarStatus.AUTH} />
+      <RegistrationForm></RegistrationForm>
+    </main>
```
main e navbar

daviromao commented on ui/src/app/(auth)/not-found/page.tsx:
```diff
@@ -1,10 +1,15 @@
-    <div className={classes.container}>
-      <NotFound></NotFound>
-    </div>
+    <main>
+      <Navbar status={NavbarStatus.AUTH}></Navbar>
+      <div className={classes.container}>
+        <NotFound></NotFound>
+      </div>
+    </main>
```
Sobre main e navbar

![image](https://github.com/user-attachments/assets/12f0b0de-fa50-4534-b59b-2f834542fc07)

daviromao commented on ui/src/app/(auth)/login/page.tsx:
```diff
@@ -1,5 +1,12 @@
 export default function LoginPage() {
-  return <LoginForm></LoginForm>;
+  return (
+    <main>
+      <Navbar status={NavbarStatus.AUTH}></Navbar>
+      <LoginForm></LoginForm>
+    </main>
```
a tag main tá relacionada ao conteúdo principal da página, em síntese seria para o texto principal, seguindo a ideia de "html semântico"

remover o <'main> da navbar e adicionar <'nav> ao conteúdo de navegação da navbar

![image](https://github.com/user-attachments/assets/e16ebdbd-4134-4820-ac15-ad986848835d)

BeaRoodriguees replied:
Então eu acho que deveria trocar para um `<div>` não? Já que a tag `<main>`  tá juntando a Navbar com o conteúdo da página.

daviromao replied:
o div não tem contexto semântico

o ideal é colocar um nav dentro do conteúdo de navegação (componente navbar)

e trazer o main apenas para o conteúdo principal (login form)

BeaRoodriguees replied:
Mas o retorno da função tem que tá em uma tag única, ai tem um componente da Navbar e outro componente com o resto da página, com qual tag eu deveria fazer esse wrap

daviromao commented on ui/src/app/page.tsx:
```diff
@@ -1,13 +1,19 @@
 
 export default function Home() {
+  const session = useSession();
+  const user = session.data?.user;
+
   return (
     <main className={classes.container}>
-      <Navbar></Navbar>
+      {user == undefined && <Navbar status={NavbarStatus.HOME} />}
+      {user && <Navbar status={NavbarStatus.LOGGED} />}
+
       <Image
         className={classes.image}
         src="/adaBackground.png"
```
Forma de verificar se está autenticado, mas já está entendível o código.
![image](https://github.com/user-attachments/assets/53309adb-e426-4ff2-94d8-32c0edb78bae)

BeaRoodriguees replied:
Desculpa, acho que não entendi o que quis dizer.

daviromao replied:
Apenas mostrei outra forma de verificar se o usuário está autenticado, não tem alterações requisitadas 😄

daviromao commented on ui/src/components/navbar/navbar.tsx:
```diff
@@ -1,23 +1,72 @@
+
   return (
     <Box>
       <header className={classes.header}>
         <Group justify="space-between" h="100%">
-          <IconLovelace className={classes.logo} />
-
-          <Group visibleFrom="sm">
-            <Button component="a" href="/register" variant="default">
-              Cadastrar
-            </Button>
-            <Button component="a" href="/login" variant="gradient" autoContrast>
-              Entrar
-            </Button>
-          </Group>
+          <Anchor href={isLogged} className={classes.link}>
+            <IconLovelace className={classes.logo} />
+          </Anchor>
```
A escolha do nome da variável "isLogged" não é uma boa, pois não dá para saber de cara qual o que ela significa.

Contudo, acho que podemos remover essa condição daqui, já que o ícone já existe um "anchor" para problemset quando está logado.

Sempre que houver um anchor em um ícone/imagem, é importante adicionar o aria-label para ele, indicando qual opção ele faz, pois um leitor de texto não consegue identificar qual o ícone

daviromao reviewed (changes requested):
Muito boa as alterações e funciona corretamente, apontei alguns problemas na questão da acessibilidade.

![image](https://github.com/user-attachments/assets/5ecf4e0f-5c71-4e4a-8785-e42aa048ab18)

BeaRoodriguees committed:
fix: improve accessibility

BeaRoodriguees committed:
fix: change tags

BeaRoodriguees committed:
fix: delete /ui/.python-version

BeaRoodriguees committed:
Merge branch 'fix/auth-pages' of https://github.com/BeaRoodriguees/lovelace into fix/auth-pages

BeaRoodriguees committed:
style: delete import

daviromao commented on ui/src/app/(auth)/login/page.tsx:
```diff
@@ -1,5 +1,12 @@
-  return <LoginForm></LoginForm>;
+  return (
+    <>
+      <Navbar status={NavbarStatus.AUTH}></Navbar>
+      <LoginForm></LoginForm>
```
```suggestion
      <Navbar status={NavbarStatus.AUTH}></Navbar>
      <main>
            <LoginForm/>
      </main>
```

daviromao commented on ui/src/app/(auth)/not-found/page.tsx:
```diff
@@ -1,10 +1,15 @@
-      <NotFound></NotFound>
-    </div>
+    <>
+      <Navbar status={NavbarStatus.AUTH}></Navbar>
+      <div className={classes.container}>
+        <NotFound></NotFound>
+      </div>
```
```suggestion
      <Navbar status={NavbarStatus.AUTH}></Navbar>
      <main className={classes.container}>
        <NotFound></NotFound>
      </main>
```

daviromao commented on ui/src/app/(auth)/register/page.tsx:
```diff
@@ -1,5 +1,12 @@
 export default function RegisterPage() {
-  return <RegistrationForm></RegistrationForm>;
+  return (
+    <>
+      <Navbar status={NavbarStatus.AUTH} />
+      <RegistrationForm></RegistrationForm>
+    </>
```
```suggestion
    <>
      <Navbar status={NavbarStatus.AUTH} />
      <main>
            <RegistrationForm/>
      </main>
    </>
```

daviromao commented on ui/src/app/page.tsx:
```diff
@@ -1,13 +1,19 @@
   return (
-    <main className={classes.container}>
-      <Navbar></Navbar>
+    <div className={classes.container}>
+      {user == undefined && <Navbar status={NavbarStatus.HOME} />}
+      {user && <Navbar status={NavbarStatus.LOGGED} />}
+
```
```suggestion
    {user == undefined && <Navbar status={NavbarStatus.HOME} />}
    {user && <Navbar status={NavbarStatus.LOGGED} />}
    <main className={classes.container}>

```

daviromao commented on ui/src/components/navbar/navbar.tsx:
```diff
@@ -1,23 +1,59 @@
+export default function Navbar({ status }: { status?: NavbarStatus }) {
   return (
     <Box>
       <header className={classes.header}>
```
Usar nav na navbar ao invés de header, apenas sugestão abaixo, alterar para que faça sentido
```suggestion
      <nav className={classes.header}>
```

BeaRoodriguees committed:
Apply suggestions from code review

Co-authored-by: Davi Romão <53953664+daviromao@users.noreply.github.com>

BeaRoodriguees committed:
style: fix lint

BeaRoodriguees committed:
style: changes requested in review

BeaRoodriguees committed:
style: 10th try
